### DISCUSSION TITLE
Como o designer entrega a interface para vocês?

### DISCUSSION THREAD

**Discussion by danilovaz** [frontendbr/forum#144] [Discussão]
Eu sei que esse é um assunto muito amplo. E muitas vezes, rola até um debate intenso disso. Mas já que estamos aqui para debater, why not?

Minha jornada de Front-End**er** é muito recente, mas como Dev Web já vão aí uns 5 anos. Nesse período eu trombei com os mais diversos designers: _fanboy de Coreldraw_, _multiplicador de layers do Photoshop_, _vetor é nóis Illustrator_, _te mandei um png_ e o _sei HTML e CSS_. Essa treta (na época) de Designer vs Desenvolvedor era muito mais sangrenta do que Homem de Ferro vs Capitão América. Já vi dev chamar o designer para resolver as paradas como "homem" porque ele inventou gradiente e borda arredondada. Cheguei até a abordar a respeito desse lance de dev vs designer em um post [aqui](https://medium.com/@danilovaz/o-designer-e-seu-amigo-nao-seu-inimigo-f4ce595505b9#.t1g6g59xk)

Eram duas profissões que se complementavam (e ainda se complementam) e estavam em processo de reconhecimento de sua importância em interfaces modernas. Quem aqui não se lembra de ter que usar o [CSS3Pie](http://css3pie.com/) pra poder ter as propriedades do CSS3 nos navegadores mais usados na época IE 8-9? Quem nunca levou esporro por defender que o usuário deveria ter um navegador atualizado? Quem nunca viu uma vaga de recortador de PSD?

Enfim, os tempos mudaram. Hoje, no front temos liberdade para testar muitas funcionalidades que ainda são draft. Com isso, cria-se possibilidade de ter interfaces realmente modernas e audaciosas. O front evoluiu e continua evoluindo bastante. But e no design?

Ainda hoje vejo muitos lugares que ainda utilizam Photoshop para criar as interfaces. E na minha opinião, Photoshop é para edição de imagens, criação de peças publicitárias e só. Em nada tem a ver com interfaces. Um PSD de um web app fica sinistro de tão pesado. Se o dev usa linux, tá lascado. Se usa VM então, senta e chora. Fora que demora muito você ir em layer por layer e ainda tem que torcer pra ser um designer organizado.

No meu caso, há pouco tempo atrás costumavam me entregar através do Sketch. E hoje na empresa utilizamos o [Zeplin.io](https://zeplin.io). Segundo os designers, agilizou muito mais o processo de criação, permitiu criar padrões e o conceito de reutilização. Geralmente também me entregam a interface com as devidas interações simuladas no [Invision](https://www.invisionapp.com/)

Como é esse processo pra vocês? Qual é a experiência de vocês com isso? Como os designers entregam as interfaces para vocês?

**Comment by icaromh** [comment]
> Como é esse processo pra vocês? Qual é a experiência de vocês com isso? Como os designers entregam as interfaces para vocês?

@danilovaz Sofrida

Lá onde trabalho o designer usa Fireworks, fica BEM mais leve que n Photoshop, mas cara... eu uso linux. É um parto, preciso subir uma VM e obviamente não fica a mesma coisa. 

Ultimamente quando recebo PSD (freela) eu tenho usado o Avocode e, essa semana vou testar o brackets.io com o lance de integração com o Adobe CC. Daí tá indo lindamente e sem nenhuma morte :dancer:

**Comment by danilovaz** [comment]
@icaromh :joy: :joy: :joy: sofrida é o que define.

É então, ferramentas que te forçam a usar SO X ou Y não tem como. Tem que ser ferramenta SaaS. Eu tenho a seguinte opinião de que o design de interfaces hoje (no geral) está no mesmo pé em que nós estávamos entre 2010-2013 e ainda tem muita gente que reluta em buscar novas formas (mais práticas, organizadas e reutilizáveis) de se criar interfaces. Aí iniciativas como o FrontUX são uma excelente oportunidade dos dois lados entenderem as dores, discutirem processos, ferramentas e focarem no que realmente importa: o usuário

**Comment by rberts** [comment]
@danilovaz Concordo cara, tem muita gente trabalhando de formas "aleatórias", haha. Eu mesmo recebo muito .psd (rola alguns .ai também) com camadas bagunçadas, até algum tempo atrás ainda recebia sem grid inclusive, tive que "educar" o designer. Com isso até me encorajou de criar uma "documentação" com algumas boas práticas para se criar interface, vou colocar aqui depois (https://github.com/istiu/layout-favoravel).

Essa questão do Zeplin é genial, mas se eu não estiver enganado ele só rola com o sketch né? O que força o cara a usar Mac. Mas tem muita ferramenta e conteúdo novo surgindo e acredito que em pouco tempo isso vai melhorar, haha :)
